### ISSUE TITLE
Feature request: Exit with non-zero status code when warnings occour

### ISSUE THREAD

**Issue by kastermester** [palantir/tslint#2604]
I would like to have a TSLint setup where most linting errors are reported as warnings (mainly due to editor integrations to be able to visually tell TS errors and linting errors apart). However when running our build system I would like to know if any warnings or errors occurred at all. Currently if only warnings occur then tslint CLI exits with status code `0`.

Would it be possible to get a command line switch or something like this to change this behavior?

**Comment by adidahiya** [comment]
I would prefer to retain consistent & predictable semantics where error == non-zero exit and warning == zero exit. A better way to approach this might be to override the `defaultSeverity` when running your build, either through a CLI flag or a separate tslint.json:

```json
{
  defaultSeverity: "warning",
  extends: "editor-tslint.json"
}
```

we'd need to fix #2569 for the latter, though.

**Comment by kastermester** [comment]
That would also work, although I can't see the harm (other than added feature set to maintain obviously) in simply adding a switch to always return non-zero if any warnings or errors occurred during the lint run.

**Comment by Buri** [comment]
I'm personally very much in favor of --strict or --max-warnings implementation.

**Comment by JoshuaKGoldberg** [comment]
Now that `defaultSeverity` can be overridden, it looks like `--max-warnings` is the only part of this issue still in discussion.

**Comment by mmajko** [comment]
bump

I agree with any other method than creating dedicated tsconfig file for failing on warning.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_
